client/auth/login.jsx

```diff
@@ -134,47 +163,66 @@ module.exports = React.createClass( {
 							<Gridicon icon="user"/>
 							<FormTextInput
 								name="login"
-								ref="login"
 								disabled={ requires2fa || inProgress }
 								placeholder={ this.translate( 'Username or email address' ) }
 								onFocus={ this.recordFocusEvent( 'Username or email address' ) }
-								valueLink={ this.linkState( 'login' ) } />
+								value={ this.state.login }
+								onChange={ this.updateLogin } />
 						</div>
 						<div className="auth__input-wrapper">
 							<Gridicon icon="lock" />
 							<FormPasswordInput
 								name="password"
-								ref="password"
 								disabled={ requires2fa || inProgress }
 								placeholder={ this.translate( 'Password' ) }
 								onFocus={ this.recordFocusEvent( 'Password' ) }
 								hideToggle={ requires2fa }
 								submitting={ inProgress }
-								valueLink={ this.linkState( 'password' ) } />
+								value={ this.state.password }
+								onChange={ this.updatePassword } />
 						</div>
-						{ requires2fa &&
+						{ ( required2faType === 'code' ) &&
 							<FormFieldset>
 								<FormTextInput
 									name="auth_code"
 									type="number"
-									ref="auth_code"
+									ref={ this.focusWhenIdle }
```

Comment:
@gyoshev figuring how to describe this is probably a worthwhile activity then. It still feels like your goal is to listen for input events, then focus that text field when no input has been received for some defined window of time. In contrast, it _looks_ like it is merely setting focus when the component first renders based on some property of the authorization process.